### PR TITLE
Turn exercise 023 to practice

### PR DESCRIPTION
**Pull request by Ozyugoo**

### CODE REVIEW

**Comment by Ozyugoo** [comment]
Hi, @cuihtlauac kindly check if this is what you asked me to do.

File: practice/023/answer/impl.ml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  let extract_rand list len =
+    extract [] (Random.int len) list
+  in
+  let rec aux n acc list len =
+    if n = 0 then acc else
+      let picked, rest = extract_rand list len in
+      aux (n - 1) (picked :: acc) rest (len - 1)
+  in
+  let len = List.length list in
+    aux (min n len) [] list len
```
Comment (cuihtlauac) [inline practice/023/answer/impl.ml]:
```suggestion
    aux (min n len) [] list len

```

File: practice/023/ex.ml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+module Make(Tested: Testable) : sig val v : test end = struct
+  let v = "rand_select" >::: [
+    "nil" >:: (fun _ -> assert_equal [] (Tested.rand_select [] 0));
+    "cons" >:: (fun _ -> assert_equal ["e"; "c"; "g"] (Tested.rand_select ["a"; "b"; "c"; "d"; "e"; "f"; "g"; "h"] 6));
+  ]
+end
+
+module Work : Testable = Work.Impl
+module Answer : Testable = Answer.Impl
```
Comment (cuihtlauac) [inline practice/023/ex.ml]:
```suggestion
module Answer : Testable = Answer.Impl

```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: practice/023/ex.ml (hunk 2 above)
Comment (Ozyugoo) [reply practice/023/ex.ml]: Thank you, on it now. I will also check the other ones and make the necessary changes

File: practice/023/answer/impl.ml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
```
Comment (cuihtlauac) [inline practice/023/answer/impl.ml]: End of file

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Tests are failing :-(

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
`dune test --root . answer` is failing

**Comment by Ozyugoo** [comment]
> Tests are failing :-(

Hey @cuihtlauac please look at this, kindly point me to the right direction

**Comment by Ozyugoo** [comment]
> `dune test --root . answer` is failing

I know, but I don't exactly understand what am doing wrong.

**Comment by cuihtlauac** [comment]
> > `dune test --root . answer` is failing
> 
> I know, but I don't exactly understand what am doing wrong

One of the expected results in the test is wrong. Probably a typo.

**Comment by Ozyugoo** [comment]
> > > `dune test --root . answer` is failing
> > 
> > 
> > I know, but I don't exactly understand what am doing wrong
> 
> One of the expected results in the test is wrong. Probably a typo.

Thank you @cuihtlauac I have seen it and it is good to go now

**Review by cuihtlauac — APPROVED** [review]

**Pull request merged**